File: src/main/java/com/unireservas/dao/ClienteDAO.java
package com.unireservas.dao;

import com.unireservas.model.Cliente;
import com.unireservas.util.DBConnection;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;

public class ClienteDAO {

    public List<Cliente> listarTodos() throws SQLException {
        List<Cliente> clientes = new ArrayList<>();
        String sql = "SELECT id_cliente, nome, telefone, email FROM CLIENTE";
        try (Connection conn = DBConnection.getConnection();
             PreparedStatement stmt = conn.prepareStatement(sql);
             ResultSet rs = stmt.executeQuery()) {
            while (rs.next()) {
                Cliente c = new Cliente();
                c.setId(rs.getInt("id_cliente"));
                c.setNome(rs.getString("nome"));
                c.setTelefone(rs.getString("telefone"));
                c.setEmail(rs.getString("email"));
                clientes.add(c);
            }
        }
        return clientes;
    }

    public Cliente buscarPorId(int id) throws SQLException {
        String sql = "SELECT id_cliente, nome, telefone, email FROM CLIENTE WHERE id_cliente = ?";
        try (Connection conn = DBConnection.getConnection();
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setInt(1, id);
            ResultSet rs = stmt.executeQuery();
            if (rs.next()) {
                return new Cliente(
                        rs.getInt("id_cliente"),
                        rs.getString("nome"),
                        rs.getString("telefone"),
                        rs.getString("email")
                );
            }
        }
        return null;
    }

    public void inserir(Cliente cliente) throws SQLException {
        String sql = "INSERT INTO CLIENTE (nome, telefone, email) VALUES (?, ?, ?)";
        try (Connection conn = DBConnection.getConnection();
             PreparedStatement stmt = conn.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
            stmt.setString(1, cliente.getNome());
            stmt.setString(2, cliente.getTelefone());
            stmt.setString(3, cliente.getEmail());
            stmt.executeUpdate();
            ResultSet rs = stmt.getGeneratedKeys();
            if (rs.next()) {
                cliente.setId(rs.getInt(1));
            }
        }
    }

    public void atualizar(Cliente cliente) throws SQLException {
        String sql = "UPDATE CLIENTE SET nome=?, telefone=?, email=? WHERE id_cliente=?";
        try (Connection conn = DBConnection.getConnection();
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, cliente.getNome());
            stmt.setString(2, cliente.getTelefone());
            stmt.setString(3, cliente.getEmail());
            stmt.setInt(4, cliente.getId());
            stmt.executeUpdate();
        }
    }

    public void deletar(int id) throws SQLException {
        String sql = "DELETE FROM CLIENTE WHERE id_cliente=?";
        try (Connection conn = DBConnection.getConnection();
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setInt(1, id);
            stmt.executeUpdate();
        }
    }
}

File: dao/ClienteDAO.js
const { query, pool } = require('../config/database');

class ClienteDAO {
  async listar() {
    return query('SELECT * FROM CLIENTE');
  }

  async listarDisponiveis() {
    return query(`SELECT * FROM CLIENTE WHERE id_cliente NOT IN (SELECT cliente_id FROM RESERVA)`);
  }

  async inserir({ nome, telefone, email }) {
    const [result] = await pool.execute('INSERT INTO CLIENTE (nome, telefone, email) VALUES (?, ?, ?)', [nome, telefone, email]);
    return { id_cliente: result.insertId, nome, telefone, email };
  }

  async atualizar(id, { nome, telefone, email }) {
    await query('UPDATE CLIENTE SET nome = ?, telefone = ?, email = ? WHERE id_cliente = ?', [nome, telefone, email, id]);
    return { id_cliente: id, nome, telefone, email };
  }

  async remover(id) {
    await query('DELETE FROM CLIENTE WHERE id_cliente = ?', [id]);
  }
}

module.exports = new ClienteDAO();
